README.md

```diff
@@ -261,6 +261,52 @@ are not there.
 
 This makes laziness work right.
 
+### handling end, error, and abort.
+
+in pull streams, any part of the stream (source, sink, or through)
+may terminate the stream. (this is the case with node streams too,
+but it's not handled well).
+
+#### source: end, error
+
+A source may end (`cb(true)` after read) or error (`cb(error)` after read)
+After ending, the source *must* never `cb(null, data)`
+
+#### sink: abort
+
+Sinks do not normally end the stream, but if they decide they do
+not need any more data they may "abort" the source by calling `read(true, cb)`.
+A abort (`read(true, cb)`) may be called before a preceding read call
```

Comment:
maybe a code example would help illustrate that sentence?
   
  function first(read, cb) {
    read(null, cb)
    read(true)
  }

Okay, and here we have the case, where read() is potentially called after a previous read indicated abort/error. So, even in this case of standard-conform behaviour, we need the robustness we talked about [here](https://github.com/dominictarr/pull-stream/commit/a95d08805ec428b707a07054e7d34d4f60e4299d#commitcomment-13148136)